### ISSUE TITLE
on('transaction') vs on('update')

### ISSUE THREAD

**Issue by Benny739** [sibiraj-s/ngx-tiptap#32]
Is there a reason why the editor directive uses on('transaction') and not on('update')?
The problem with the transaction event is, that setContent always triggers it with docChanged = true. 
With the update event it gets only triggered when something inside the editor changes. 

The consequence it that patchValue on a pristine form marks the form as dirty, because the editor emits a value.

In my test the update event worked perfectly fine.

**Comment by sibiraj-s** [comment]
To be frank, I don't remember why I made the change. Also, I don't remember seeing an update event. At that time most of their docs were in progress. Or I could not have noticed/missed.

I am open to making this change. But, unless we properly evaluate the difference between the two events. I think it is worth considering it as a breaking change. Thoughts?

**Comment by Benny739** [comment]
Yeah we could consider it a breaking change. 
According to the tiptap docs update is the proper event here. 
I tested it and I could not see any difference. On transaction just emits way more often but when it comes to the content change there was no difference.
I think we could go with a minor version and mention the change in the behavior.

Should I create a PR?

**Comment by sibiraj-s** [comment]
> Should I create a PR?

Sure.

**Comment by sibiraj-s** [comment]
And I think I remember the reason for using the transaction event. There are certain commands where update events are not triggered. But I think there is a workaround, all commands of this category we need to add an extra boolean parameter to manually update them.

Example:

Before

```js
editor.commands.setContent(‘Hello World!')
```

After

```js
editor.commands.setContent(‘Hello World!’, true)
```

Ref: https://tiptap.dev/api/commands/set-content

**Comment by sibiraj-s** [comment]
And need to verify the `OnPush` changedetection strategy as well.

**Comment by sibiraj-s** [comment]
Note: Moving from transaction to update event might fix #24

**Comment by Benny739** [comment]
Yes the emitUpdate option is actually important, because #24 arises from onTransaction not respecting the emitUpdate

```
this.editor.chain().setContent(value, false).run();
```
We already have emitUpdate = false in writeValue, so it shouldn't trigger an update, but only on('update') respects that option.

OnPush is no problem here, because handleChange isn't changing. The only difference the on('update') should do, is not emit a value in the writeValue function.

**Comment by sibiraj-s** [comment]
Fixed in v7.0.0
